### PR TITLE
EZP-27286: Fix BDD to work with content browse

### PR DESCRIPTION
**Pull request by StephaneDiot**
JIRA : https://jira.ez.no/browse/EZP-27286
## Description

With content tree being removed BDD tests are not up-to-date. I'm adding some tests to work with the new content browse.

### CODE REVIEW

File: Features/ContentActions/MoveContent.feature [hunk 1]
```diff
@@ -18,12 +18,12 @@ Feature: Move content
         And I select the "eZ Platform/Older News" folder in the Universal Discovery Widget
         And I confirm the selection
         Then I am notified that "News Flash" has been moved under "Older News"
-        And I see "Older News/News Flash" in the content tree
-        And I do not see "Origin/News flash" in the content tree
+        And There is a content "Older News/News Flash"
```
Comment (dpobel) [inline Features/ContentActions/MoveContent.feature]: I would write
```feature
And "Older News/News Flash" content item exists
And "Origin/News flash" content item does not exist
```

File: Features/ContentActions/CopyContent.feature [hunk 2]
```diff
@@ -18,7 +18,7 @@ Feature: Copy content
         And I select the "eZ Platform/Destination" folder in the Universal Discovery Widget
         And I confirm the selection
         Then I am notified that "News Flash" has been copied under "Destination"
-        And I see "Destination/News Flash" in the content tree
+        And There is a content "Destination/News Flash"
```
Comment (bdunogier) [inline Features/ContentActions/CopyContent.feature]: lowercase "T" on there :)

File: Features/ContentActions/MoveContent.feature (hunk 1 above)
Comment (bdunogier) [inline Features/ContentActions/MoveContent.feature]: I'd even add alternatives to those sentences, even if the implementation is identical:
- `the content item "Older News/Tomorrow news/News Flash" was (not) removed`
- `the content item "Older News/Tomorrow news/News Flash" was (not) sent to trash`
- `the content item was moved to "Older News/Tomorrow news/News Flash"`

They can all use the same `theContentItemExists()` implementation, but will sound better in the scenarii.

File: Features/Context/SubContext/CommonActions.php [hunk 3]
```diff
@@ -231,17 +231,45 @@ public function clickEditActionBar($button)
...
-     * Explores the content tree, expanding it and click on the desired element
+     * @Then There is not a content :path
+     * @Then There isn't a content :path
+     * Explores the finder of the UDW, verify the desired element doesn't exist and close the UDW.
      *
-     * @param   string  $path    The content tree path such as 'Content1/Content2/ContentIWantToClick'
+     * @param   string  $path    The content browse path such as 'Content1/Content2/ContentIWantToClick'
      */
-    public function clickOnTreePath($path)
+    public function verifyContentIsMissing($path)
```
Comment (bdunogier) [inline Features/Context/SubContext/CommonActions.php]: The method's name should be adapted to be closer to what is checked (`function thereIsNoContent($path)`) (including the sentence change suggested by Damien).

File: Features/Context/SubContext/CommonActions.php [hunk 4]
```diff
@@ -231,17 +231,45 @@ public function clickEditActionBar($button)
...
+        $this->clickNavigationItem('Content structure');
+        $this->dontSeeBrowsePath($path);
+        $this->cancelSelection();
+    }
+
+    /**
+     * @Then There is a content :path
+     * Explores the finder of the UDW, find the desired element and close the UDW.
+     *
+     * @param   string  $path    The content browse path such as 'Content1/Content2/ContentIWantToClick'
```
Comment (bdunogier) [inline Features/Context/SubContext/CommonActions.php]: No need to add the extra spaces, we don't align statements. And there is nothing to align with :)

File: Features/Context/SubContext/CommonActions.php [hunk 5]
```diff
@@ -231,17 +231,45 @@ public function clickEditActionBar($button)
...
+     */
+    public function verifyContent($path)
```
Comment (bdunogier) [inline Features/Context/SubContext/CommonActions.php]: `verifyContent` isn't a good method name. `public function thereIsAContent()` is better.

File: Features/Context/SubContext/CommonActions.php [hunk 6]
```diff
@@ -231,17 +231,45 @@ public function clickEditActionBar($button)
...
+    {
+        $this->clickNavigationItem('Content structure');
+        $this->clickOnBrowsePath($path);
+        $this->confirmSelection();
+    }
+
+    /**
+     * Explores the UDW, expanding it and click on the desired element.
+     *
+     * @param   string  $path    The content browse path such as 'Content1/Content2/ContentIWantToClick'
```
Comment (bdunogier) [inline Features/Context/SubContext/CommonActions.php]: Extra spaces.

File: Features/Context/SubContext/CommonActions.php [hunk 7]
```diff
@@ -287,11 +315,22 @@ public function selectFromUniversalDiscovery($path)
...
-        $this->clickElementByText('Confirm selection', '.ez-universaldiscovery-confirm');
+        $elem = $this->findWithWait('.ez-view-universaldiscoveryview');
+        $elem->find('css', '.ez-universaldiscovery-confirm')->click();
+    }
+
+    /**
+     * @When I cancel the selection
+     * Cancel the selection in Universal discovery.
+     */
+    public function cancelSelection()
```
Comment (bdunogier) [inline Features/Context/SubContext/CommonActions.php]: A bit bothered by the fact that UDW sentences and methods are part of `CommonActions`. Could you try to change that ?

File: .travis.yml [hunk 8]
```diff
@@ -14,7 +14,7 @@ cache:
 
 matrix:
     fast_finish: true
-    allow_failures:
+    include:
```
Comment (yannickroger) [inline .travis.yml]: This PR should be opened against 1.9 as you allowed failures there in the first place.

**Comment by StephaneDiot** [comment]
updated

**Comment by dpobel** [comment]
still ok for me :)
@bdunogier could you have another look please ?

**Pull request merged**